data_model/src/evaluate.rs

```diff
@@ -540,87 +540,74 @@ mod tests {
-                        ContextValue::new(Name::from_str("usd_quantity").expect("Can't fail."))
-                            .into(),
-                    ),
+                    EvaluatesTo::new_unchecked(ContextValue::new(
+                        Name::from_str("usd_quantity").expect("Can't fail."),
+                    )),
                     1000_u32,
                 ),
             ),
-            EvaluatesTo::new_evaluates_to_value(
-                Or::new(
-                    ContainsAll::new(
-                        EvaluatesTo::new_unchecked(
-                            ContextValue::new(Name::from_str("signatories").expect("Can't fail."))
-                                .into(),
-                        ),
-                        teller_signatory_set.clone(),
-                    ),
-                    Contains::new(
-                        EvaluatesTo::new_unchecked(
-                            ContextValue::new(Name::from_str("signatories").expect("Can't fail."))
-                                .into(),
-                        ),
-                        manager_signatory,
-                    ),
-                )
-                .into(),
-            ),
+            EvaluatesTo::new_evaluates_to_value(Or::new(
+                ContainsAll::new(
+                    EvaluatesTo::new_unchecked(ContextValue::new(
+                        Name::from_str("signatories").expect("Can't fail."),
+                    )),
+                    teller_signatory_set.clone(),
+                ),
+                Contains::new(
+                    EvaluatesTo::new_unchecked(ContextValue::new(
+                        Name::from_str("signatories").expect("Can't fail."),
+                    )),
+                    manager_signatory,
+                ),
+            )),
             true,
         );
         // Signed by all tellers
-        let expression = Where::new(EvaluatesTo::new_evaluates_to_value(
-            condition.clone().into(),
-        ))
-        .with_value(
-            //TODO: use query to get the actual quantity of an asset from WSV
-            // in that case this test should be moved to iroha_core
-            Name::from_str("usd_quantity").expect("Can't fail."),
-            asset_quantity_high.clone(),
-        )
-        .with_value(
-            Name::from_str("signatories").expect("Can't fail."),
-            teller_signatory_set,
-        );
+        let expression = Where::new(EvaluatesTo::new_evaluates_to_value(condition.clone()))
+            .with_value(
+                //TODO: use query to get the actual quantity of an asset from WSV
+                // in that case this test should be moved to iroha_core
+                Name::from_str("usd_quantity").expect("Can't fail."),
+                asset_quantity_high.clone(),
+            )
+            .with_value(
+                Name::from_str("signatories").expect("Can't fail."),
+                teller_signatory_set,
+            );
```

Comment:
No semantic changes. Multiple places. We might want to consider updating this API as well. `EvaluatesTo::new_evaluates_to_value` seems awfully redundant.